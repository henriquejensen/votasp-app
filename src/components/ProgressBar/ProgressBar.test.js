import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  it('should renders correctly when values is undefined', () => {
    expect(toJson(shallow(( <ProgressBar /> )))).toMatchSnapshot();
  })

  it('should renders correctly when values exists and IS sorted', () => {
    const values = [
      { color: 'green', value: 100 },
      { color: 'yellow', value: 50 },
      { color: 'red', value: 0 }
    ];
    expect(toJson(shallow(( <ProgressBar values={values} /> )))).toMatchSnapshot();
  })

  it('should renders correctly when values exists and IS NOT sorted', () => {
    const values = [
      { color: 'red', value: 0 },
      { color: 'yellow', value: 50 },
      { color: 'green', value: 100 }      
    ];
    expect(toJson(shallow(( <ProgressBar values={values} /> )))).toMatchSnapshot();
  })
})
