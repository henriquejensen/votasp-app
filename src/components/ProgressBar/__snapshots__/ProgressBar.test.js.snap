// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should renders correctly when values exists and IS NOT sorted 1`] = `
<div
  className="progress-bar-container"
>
  <svg
    className="view-port"
    viewBox="-4 0 108 10"
  >
    <line
      stroke="white"
      strokeLinecap="round"
      strokeWidth="4"
      x1="0"
      x2="100"
      y1="5"
      y2="5"
    />
    <line
      className="fill-bar"
      key="0"
      stroke="green"
      strokeLinecap="round"
      strokeWidth="2"
      x1="0"
      x2={100}
      y1="5"
      y2="5"
    />
    <line
      className="fill-bar"
      key="1"
      stroke="yellow"
      strokeLinecap="round"
      strokeWidth="2"
      x1="0"
      x2={50}
      y1="5"
      y2="5"
    />
    <line
      className="fill-bar"
      key="2"
      stroke="red"
      strokeLinecap="round"
      strokeWidth="2"
      x1="0"
      x2={0}
      y1="5"
      y2="5"
    />
  </svg>
</div>
`;

exports[`ProgressBar should renders correctly when values exists and IS sorted 1`] = `
<div
  className="progress-bar-container"
>
  <svg
    className="view-port"
    viewBox="-4 0 108 10"
  >
    <line
      stroke="white"
      strokeLinecap="round"
      strokeWidth="4"
      x1="0"
      x2="100"
      y1="5"
      y2="5"
    />
    <line
      className="fill-bar"
      key="0"
      stroke="green"
      strokeLinecap="round"
      strokeWidth="2"
      x1="0"
      x2={100}
      y1="5"
      y2="5"
    />
    <line
      className="fill-bar"
      key="1"
      stroke="yellow"
      strokeLinecap="round"
      strokeWidth="2"
      x1="0"
      x2={50}
      y1="5"
      y2="5"
    />
    <line
      className="fill-bar"
      key="2"
      stroke="red"
      strokeLinecap="round"
      strokeWidth="2"
      x1="0"
      x2={0}
      y1="5"
      y2="5"
    />
  </svg>
</div>
`;

exports[`ProgressBar should renders correctly when values is undefined 1`] = `
<div
  className="progress-bar-container"
>
  <svg
    className="view-port"
    viewBox="-4 0 108 10"
  >
    <line
      stroke="white"
      strokeLinecap="round"
      strokeWidth="4"
      x1="0"
      x2="100"
      y1="5"
      y2="5"
    />
  </svg>
</div>
`;
